src/go/rpk/pkg/cli/transform/init.go

```diff
@@ -0,0 +1,257 @@
+	return err
+}
+
+func (*optionalBool) Type() string {
+	return "bool"
+}
+
+func (b *optionalBool) String() string {
+	if b.ok == nil {
+		return "prompt"
+	}
+	return strconv.FormatBool(*b.ok)
+}
+
+func (*optionalBool) IsBoolFlag() bool { return true }
+
+func newInitializeCommand(fs afero.Fs) *cobra.Command {
+	var (
+		name    string
+		lang    project.WasmLang
+		install optionalBool
+	)
+	cmd := &cobra.Command{
+		Use:   "init [DIRECTORY]",
+		Short: "Initialize a transform",
+		Long: `Initialize a transform.
+
+Creates a new transform using a template in the current directory.
+
+A new directory can be created by specifying it in the command, like:
+
+rpk transform init foobar
+
+Will initialize a transform project in the foobar directory.
+		`,
+		Args: cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			path, err := computeProjectDirectory(fs, args)
+			out.MaybeDie(err, "unable to determine project directory: %v", err)
+			c := filepath.Join(path, project.ConfigFileName)
+			ok, err := afero.Exists(fs, c)
+			out.MaybeDie(err, "unable to determine if %q exists: %v", c, err)
+			if ok {
+				out.Die("there is already a transform at %q, please delete it before retrying", c)
+			}
+			for name == "" {
+				suggestion := filepath.Base(path)
+				if suggestion == "." {
+					suggestion = ""
+				}
+				name, err = out.PromptWithSuggestion(suggestion, "name this transform:")
+				out.MaybeDie(err, "unable to determine project name: %v", err)
+			}
+			if lang == "" {
+				langVal, err := out.Pick(project.AllWasmLangs, "select a language:")
+				out.MaybeDie(err, "unable to determine transform language: %v", err)
+				lang = project.WasmLang(langVal)
+			}
+			p := transformProject{Name: name, Lang: lang, Path: path}
+
+			fmt.Printf("generating project in %s...\n", p.Path)
+			err = executeGenerate(fs, p)
+			out.MaybeDie(err, "unable to generate all manifest files: %v", err)
+			fmt.Println("created project in", p.Path)
+
+			if install.ok == nil {
+				ok, err = out.Confirm("install dependencies?")
+				install.ok = &ok
+				out.MaybeDie(err, "unable to determine if dependencies should be installed: %v", err)
+			} else if *install.ok {
+				fmt.Println("installing dependencies...")
+			}
+			if *install.ok {
+				err = installDeps(cmd.Context(), fs, p)
+				out.MaybeDie(err, "unable to install dependencies: %v", err)
+				fmt.Println("dependencies installed")
+			}
+
+			fmt.Println("deploy your transform using:")
+			cwd, err := os.Getwd()
+			out.MaybeDie(err, "unable to get current directory: %v", err)
+			if cwd != path {
+				rel, err := filepath.Rel(cwd, path)
+				if err == nil {
+					fmt.Println("\tcd", rel)
+				}
+			}
+			fmt.Println("\trpk transform build")
+			fmt.Println("\trpk transform deploy")
+		},
+	}
+	cmd.Flags().VarP(&lang, "language", "l", "The language used to develop the transform")
+	cmd.Flags().Var(&install, "install-deps", "If dependencies should be installed for the project")
```

Comment:
[question] How does this flag work? just having the flag present means that the value is `true` ? I see that in the Set method (L35) you accept a `prompt` string value? in which cases do we use that in the flag?

Reply:
It's basically a tri-state flag.

If you don't specify the flag we prompt otherwise you can do the following to skip the prompt:

`--install-deps` (auto install without prompt)
`--install-deps=true` (same as above)
`--install-deps=false` (don't install and don't prompt)

The theme here is that if you don't specify a flag value we prompt you for input. I'm happy to always install dependencies without prompting the user and having this flag be `--no-install-deps`. Thoughts?

Reply:
I think I like how it's written at the moment.

I had a question about how the flag worked because I'm not familiar with flags with custom types. But now I understand it. Thanks :+1: